react/features/invite/components/InviteButton.web.js

```diff
@@ -0,0 +1,231 @@
+                        shouldFitContainer = { true }>
+                        { t('invite.invitePeople') }
+                    </Button>
+                    { this.props._isDialOutAvailable
+                        || this.props._isAddToCallAvailable
+                        ? <DropdownMenu
+                            items = { this.state.inviteOptions }
+                            onItemActivated = { this._onInviteOptionSelected }
+                            position = { VERTICAL_FILMSTRIP
+                                ? 'bottom right'
+                                : 'top right' }
+                            shouldFlip = { true }
+                            triggerType = 'button' />
+                        : null }
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Handles the click of the invite button.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onInviteClick() {
+        this.props.openDialog(InviteDialog);
+    }
+
+    /**
+     * Handles selection of the invite options.
+     *
+     * @param { Object } option - The invite option that has been selected from
+     * the dropdown menu.
+     * @private
+     * @returns {void}
+     */
+    _onInviteOptionSelected(option) {
+        this.state.inviteOptions[0].items.forEach(item => {
+            if (item.content === option.item.content) {
+                item.action();
+            }
+        });
+    }
+
+    /**
+     * Updates the invite items list depending on the availability of the
+     * features.
+     *
+     * @param {Object} props - The read-only properties of the component.
+     * @private
+     * @returns {void}
+     */
+    _updateInviteItems(props) {
+        const { t } = this.props;
+
+        const inviteItems = [];
+
+        inviteItems.splice(
+            getInviteOptionPosition(SHARE_LINK_OPTION),
+            0,
+            {
+                content: t('toolbar.invite'),
+                action: () => this.props.openDialog(InviteDialog)
+            }
+        );
+
+        if (props._isDialOutAvailable) {
+            inviteItems.splice(
+                getInviteOptionPosition(DIAL_OUT_OPTION),
+                0,
+                {
+                    content: t('dialOut.dialOut'),
+                    action: () => this.props.openDialog(DialOutDialog)
+                }
+            );
+        }
+
+        if (props._isAddToCallAvailable) {
+            inviteItems.splice(
+                getInviteOptionPosition(ADD_TO_CALL_OPTION),
+                0,
+                {
+                    content: interfaceConfig.ADD_PEOPLE_APP_NAME,
+                    action: () => this.props.openDialog(AddPeopleDialog)
+                }
+            );
+        }
+
+        this.state = {
+            /**
+             * The list of invite options.
+             */
+            inviteOptions: [
+                {
+                    items: inviteItems
```

Comment:
Is this state necessary? Could this be done within the scope of render?

Reply:
By using the state I can make sure that "inviteItems" array is only updated when one of the 2 properties I check in componentWillReceiveProps changes. This way the dropdown will be re-rendered only if the inviteItems has changed.

Reply:
Do you feel that optimization is necessary? What about using PureComponent then?

Reply:
Actually, I'm not sure I see what bothers you in here. I feel like this is the most straightforward react way to do such a thing.

Reply:
I generally prefer reducing the usage of member variables if it's possible to avoid them without much effort. It comes down to keeping variable access as restrictive as possible.